docs/reference/modules/asset.md

```diff
@@ -0,0 +1,205 @@
+
+#### `extensions` function
+
+Functions in the `extensions` property take an `option` parameter to allow multiple modules to modify the same extension through the [`extensionOptions`](#extensionoptions) property. Each property to be altered across multiple modules must be present in the option that was originally returned.
+
+<AposCodeBlock>
+
+```javascript
+const path = require("path");
+
+module.exports = {
+ // ...
+  webpack: {
+   extensions: {
+     addAlias(options) {
+       return {
+         resolve: {
+           extensions: [ ...(options.extensions || []), '.jsx' ],
+           alias: {
+             ...(options.alias || {}),
+             Utilities: path.resolve(process.cwd(), './lib/utils')
+           }
+         }
+       };
+     }
+   }
+ },
+ // ...
+};
+
+```
+<template v-slot:caption>
+modules/my-module/index.js
+</template>
+</AposCodeBlock>
+
+### `extensionOptions`
+Any module can add additional functionality to an existing webpack `extension` that was created through a function using the `extensionOptions` property. This property takes a function with an `options` parameter and returns an object with additional options for any configuration options present in the original function.
+
+#### Example
+
+Extending the previous example with an additional extension and alias:
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+ // ...
+  webpack: {
+   extensionOptions: {
+     addAlias(options) {
+       return {
+         extensions: [ '.jsm' ],
+         alias: {
+           Data: path.resolve(process.cwd(), './lib/data')
+         }
+       };
+     }
+   }
+ },
+// …
+};
+```
+<template v-slot:caption>
+module/my-new-widget/index.js
+</template>
+</AposCodeBlock>
+
+### `bundles`
+The `bundles` option allows delivery of frontend assets to a specific page or if a widget is on a page page. This option takes an object containing either the JavaScript or stylesheet file names without extension as properties. These files should be located in the module's `ui/src` folder. Each property takes either an empty object or a sub-property of `templates`. The `templates` sub-property accepts an array of page template names where the content should be loaded. So, for example, a bundle added to a `piece-page-type` could specify `templates: [ 'index' ]` to load the bundle on the `index` page, but not the `show` page.
+
+Multiple modules can contribute to a single bundle by using the same bundle name, and webpack will merge these files in the final build. In addition, any shared dependencies between the extra bundles and the "main" bundle built from the code found in the `ui/src/index.js` files will be resolved so that only one instance is loaded. 
+
+::: warning
+Bundles should be used cautiously as they can contribute to slower page load times. 
```

Comment:
The warning on the webpack guide page is specific about when this is true and when it is not. You might want to bring it over verbatim.

Reply:
Done.